File: src/main/java/org/pl/lexer/token/IValueToken.java
package org.pl.lexer.token;


/**
 * Value tokens are tokens that hold an additional value.
 *
 * @param <T> Value type.
 */
public interface IValueToken<T> extends IToken {

    /**
     * Returns the value of the token.
     *
     * @return Value.
     */
    T getValue();
}

File: src/main/kotlin/org/nyxlang/lexer/token/IValueToken.kt
package org.nyxlang.lexer.token

/**
 * Value tokens are tokens that hold an additional value.
 *
 * @param <T> Value type.
 */
interface IValueToken<T> : IToken {

    /**
     * The immutable value of the token.
     */
    val value: T
}